src/lib.rs

```diff
@@ -464,6 +464,8 @@ impl fmt::Debug for PctStr {
 /// ```
 pub trait Encoder {
 	/// Decide if the given character must be encoded.
+	///
+	/// Note that the character `%` is always encoded even if this method returns `false` on it.
```

Comment:
This is not just an advice, it *must* be enforced by the `PctString::encode` function. If not, the resulting `PctString` may not be a valid percent encoded string.

For instance if we consider the string `"%21"` (not encoded) and apply `PctString::encode` on it, if `%` is not encoded then the resulting string `"%21"` is encoding `"!"` and not the original string. `%` must always be encoded, even if the `Encoder` implementation says otherwise.

Reply:
Maybe we should add that as an example in the doc.

Reply:
And a test.

Reply:
Okay. That is the text that was there previously so i just put it back.

Reply:
Yes but I think the new implementation of `PctString::encode` is not following this comment, right?

Reply:
Yeah, but I'm saying this was a simple revert 😛 this is where this line was before!

Reply:
Made the language more forceful, put it in a few places, and added two tests.